src/flagStore.ts

```diff
@@ -81,8 +104,10 @@ export class FlagStore {
 
 	async removeAllListeners(): Promise<void> {
 		try {
-			const ldClient = await this.ldClient;
-			await ldClient.removeAllListeners('update');
+			await setTimeout(async () => {
+				const ldClient = await this.ldClient;
+				await ldClient.removeAllListeners('update');
+			}, 500);
```

Comment:
What's the timeout for?

Reply:
This is an order of operations issue from what I can sort out. When the extension is first created there is no ldclient created so it could cause everything to hang. Removing listeners is really best effort for our use case too. They'll be cleaned up either way if the client itself closes out and the Treeview should always map to the current client.